app/utils/SourceFile.py

```diff
@@ -117,14 +117,14 @@ def __create_patch(self, line_number: int, replacement: Replacement) -> str:
         ))
 
         # lines: context before
-        patch_lines += [' ' + x + '\n' for x in context_before]
+        patch_lines += [' ' + x + "\r\n" for x in context_before]
```

Comment:
Why is `\r\n` needed here?

Reply:
Git can clone files with different line endings mode. By default, as I guess, git clone with ```\r\n``` and pushes with ```\n```. So on my Ubuntu 16.04 I have repositories with ```\r\n``` line endings in files. So patch didn't work with ```\n``` and works with ```\r\n```. But to be fair it's better to handle both situations...